src/openvslam/system.cc

```diff
@@ -238,6 +238,11 @@ void system::abort_loop_BA() {
     global_optimizer_->abort_loop_BA();
 }
 
+bool system::is_tracking(){
+    std::lock_guard<std::mutex> lock(mtx_tracking_);
+    ((tracker_->tracking_state_ == tracker_state_t::Tracking) ? true : false);
+}
+
```

Comment:
It looks like return is missing.

```
/openvslam/src/openvslam/system.cc: In member function 'bool openvslam::system::is_tracking()':
/openvslam/src/openvslam/system.cc:243:78: warning: second operand of conditional expression has no effect [-Wunused-value]
     ((tracker_->tracking_state_ == tracker_state_t::Tracking) ? true : false);
                                                                              ^
/openvslam/src/openvslam/system.cc:243:78: warning: third operand of conditional expression has no effect [-Wunused-value]
/openvslam/src/openvslam/system.cc:244:1: warning: no return statement in function returning non-void [-Wreturn-type]
 }
 ^
```

Consider renaming `is_tracking()` to `current_cam_pose_is_valid()`.
I think what we want to check is not whether we are tracking or not, but whether the pose is valid or not.
For example, even if you are tracking, you may not be able to estimate a valid pose for a single frame.

```suggestion
bool system::current_cam_pose_is_valid(){
    std::lock_guard<std::mutex> lock(mtx_tracking_);
    return curr_frm_.cam_pose_cw_is_valid_;
}

```

Reply:
How about including information about whether the pose is valid or not in the return value from `feed_*_frame`?
Change the location of `publish_pose()` so that it can receive the return value of `feed_*_frame` and publish it.